src/bam/collapse_reads_to_fragments/pipeline.rs

```diff
@@ -63,26 +63,51 @@ impl CallConsensusRead {
                         //Case: Right record
                         Some(record_pair) => {
                             //For right record save end position and duplicate group ID
-                            group_end_idx
-                                .entry(record.cigar_cached().unwrap().end_pos() - 1)
-                                .or_insert_with(HashSet::new)
-                                .insert(duplicate_id.integer());
+                            let mut mod_duplicate_id = duplicate_id.integer();
+                            let record_end_pos = record.cigar_cached().unwrap().end_pos() - 1;
                             match record_pair {
-                                RecordStorage::PairedRecords { ref mut r_rec, .. } => r_rec
-                                    .get_or_insert(IndexedRecord {
+                                RecordStorage::PairedRecords { ref mut r_rec, .. } => {
+                                    r_rec.get_or_insert(IndexedRecord {
                                         rec: record,
                                         rec_id: i,
-                                    }),
-                                RecordStorage::SingleRecord { .. } => unreachable!(),
+                                    });
+                                }
+                                // This arm is reached if a mate is mapped to another chromosome.
+                                // In that case a new duplicate and record ID is required
+                                RecordStorage::SingleRecord { .. } => {
+                                    mod_duplicate_id *= -1;
+                                    let mut new_record_id = record_id.to_vec();
+                                    new_record_id.push(1);
+                                    duplicate_groups
+                                        .entry(mod_duplicate_id)
+                                        .or_insert_with(Vec::new)
+                                        .push(new_record_id.clone());
+                                    record_storage.insert(
+                                        new_record_id,
+                                        RecordStorage::SingleRecord {
+                                            rec: IndexedRecord {
+                                                rec: record,
+                                                rec_id: i,
+                                            },
+                                        },
+                                    );
+                                }
```

Comment:
This should fix the issue for duplicate groups.
The difficulty is that in case a read pairs maps to different chromosomes both reads need to be assigned to different duplicate groups in order to cluster und merge those reads separately.
In ordner to do so the second read of a pair requires a different record ID and a different duplicate ID.
As the duplicate ID are integer values I decided to just negate them so there should be no collision with other IDs.
The record ID is saved as a `Vec<u8>`. For modification I just appended `1`. But in that case I'm not sure if there won't be any collisions.